code/game/machinery/_machinery.dm

```diff
@@ -148,6 +148,8 @@
 	var/always_area_sensitive = FALSE
 	///Multiplier for power consumption.
 	var/machine_power_rectifier = 1
+	/// Are we upgraded with metallic hydrogen for less power consumption?
+	var/has_hydrogen_upgrade = FALSE
```

Comment:
This really seems like you could handle this with a component or element rather than sticking a new var on base `/machine`. 

All you'd need to do is add a signal to `RefreshParts()` to modify the power usages. 

Like `/datum/element/metal_hydrogen_modifier`, which would register a deconstruct signal to return metal hydrogen and a refreshparts signal to modify power. 

Or more generally, a `/datum/component/machine_power_modifier`, which `/obj/item/stack/sheet/mineral/metal_hydrogen` applies when you attack a machine with it, and optionally takes a callback ran on application or machine deconstruct.

(If you use the general approach, you'd need a way to prevent source stacking, which would be easy. Feel free to ask about it if you don't know.)